client/src/pages/CalendarPage/CalendarPage.js

```diff
@@ -0,0 +1,127 @@
+import React, { useState } from "react";
+import "react-calendar/dist/Calendar.css";
+import "./CalendarOverride.css";
+import Calendar from "react-calendar";
+import styles from "./CalendarPage.module.css";
+import TaskCard from "../../components/TaskCard/TaskCard.js";
+import CalendarHeader from "../../components/CalendarHeader/CalendarHeader.js";
+
+const dayLabelFormatter = (locale, label) => {
+  return label.toString().slice(0, 1);
+};
+
+const dailyTaskDateFormatter = (date) => {
+  const months = [
+    "January",
+    "February",
+    "March",
+    "April",
+    "May",
+    "June",
+    "July",
+    "August",
+    "September",
+    "October",
+    "November",
+    "December",
+  ];
+  return String.raw`${
+    months[date.getMonth()]
+  } ${date.getDate()}, ${date.getFullYear()}`.toString();
+};
+
+const CalendarPage = () => {
+  const [currDate, onDateChange] = useState(new Date());
+  const children = [
+    {
+      title: "GDSC Meeting",
+      locationText: "DH2020",
+      startTimeText: "05:00PM",
+      endTimeText: "07:00PM",
+      isOngoing: false,
+    },
+    {
+      title: "another meeting",
+      locationText: "DH2020",
+      startTimeText: "05:00PM",
+      endTimeText: "07:00PM",
+      isOngoing: true,
+    },
+    {
+      title: "GDSC Meeting",
+      locationText: "DH2020",
+      startTimeText: "05:00PM",
+      endTimeText: "07:00PM",
+      isOngoing: false,
+    },
+    {
+      title: "another meeting",
+      locationText: "DH2020",
+      startTimeText: "05:00PM",
+      endTimeText: "07:00PM",
+      isOngoing: true,
+    },
+    {
+      title: "GDSC Meeting",
+      locationText: "DH2020",
+      startTimeText: "05:00PM",
+      endTimeText: "07:00PM",
+      isOngoing: false,
+    },
+    {
+      title: "another meeting",
+      locationText: "DH2020",
+      startTimeText: "05:00PM",
+      endTimeText: "07:00PM",
+      isOngoing: true,
+    },
+    {
+      title: "GDSC Meeting",
+      locationText: "DH2020",
+      startTimeText: "05:00PM",
+      endTimeText: "07:00PM",
+      isOngoing: false,
+    },
+    {
+      title: "another meeting",
+      locationText: "DH2020",
+      startTimeText: "05:00PM",
+      endTimeText: "07:00PM",
+      isOngoing: true,
+    },
+  ];
+  const cards = children.map((task, ix) => (
+    <li key={ix} style={{ listStyle: "none" }}>
+      <TaskCard
+        title={task.title}
+        locationText={task.locationText}
+        startTimeText={task.startTimeText}
+        endTimeText={task.endTimeText}
+        isOngoing={task.isOngoing}
+        showDetailsDialog={() => {
+          console.log("show the modal from here");
+        }}
+      />
+    </li>
+  ));
+
+  return (
+    <div className={styles.bg}>
+      <CalendarHeader />
+      <Calendar
+        onChange={onDateChange}
```

Comment:
The onDateChange will need to be another function that gets the new day that was clicked and from there you'll be able to set onDateChange(whateverTheNewDateIs)

I'm not sure what gets returned from the Calendar onChange, but for example, using an `Input` it'll be something like the following:

Input onChange example (with state)
```
...
const [inputText, setInputText] = useState("");

return (
    <input onChange=((e) => {setInputText(e.target.value)}) value={inputText}/>
)
// From this example, you can see that `e` gets returned from the input onChange and we know that we can extract the text using `e.target.value` (It might be different for the Calendar onChange)
```
Once again, this will depend on what the Calendar onChange returns (You can check what gets returned from a `console.log`

Reply:
Added a new function accordingly. Turns out, Calendar passes in a js `Date` object.